proof/access-control/Retype_AC.thy

```diff
@@ -418,8 +419,17 @@ end
 
 context retype_region_proofs' begin
 
-lemma domains_of_state: "domains_of_state s' \<subseteq> domains_of_state s"
-  unfolding s'_def by simp
+lemma tcb_domain_map_wellformed:
+  "\<lbrakk>tcb_domain_map_wellformed aag s; pas_cur_domain aag s; \<forall>x\<in> set (retype_addrs ptr ty n us). is_subject aag x\<rbrakk>
+   \<Longrightarrow> tcb_domain_map_wellformed aag s'"
+  unfolding s'_def ps_def
+  apply (clarsimp simp: tcb_domain_map_wellformed_aux_def)
+  apply (erule domains_of_state_aux.cases)
+  apply (clarsimp simp: etcbs_of'_def split: if_split_asm kernel_object.splits)
+   apply (clarsimp simp: default_object_def default_tcb_def tyunt split: apiobject_type.splits)
+   defer
+  apply (force intro: domtcbs)
```

Comment:
is this copied from elsewhere? post-`defer` indent is wrong, and probably there's no need to split obj and aobj separately anyway, and on top of that force can probably finish it after the erule, or maybe even do the whole lemma if we're lucky